### ISSUE TITLE
why are the modules not return in the callback?

### ISSUE THREAD

**Issue by alvarolorentedev** [EvolveLabs/electron-plugins#4]
so i have my use case where i need to call some functions common to my modules during execution. I am finding that this seems not actually possible as in the callback the return values are a list of the 'dependencies'. 

``` js
function loadPlugin(context, results, callback) {
    var modules = []
    var dependencies = []
    try {
            ...
            var Plugin = require(file)
            var mod = new Plugin(context.appContext)
            modules.push(mod)
            dependencies.push(depName)
        }
    } catch (err) {
        return callback(err)
    }

    callback(null, dependencies)
}
```

is there any why the modules collection should not be returned in the callback?

**Comment by justinmchase** [comment]
That would probably be ok but you can also add things onto the `context.appContext`.

```
var appContext = {
  common: function (dep) { console.log(dep); }
}

loadPlugins({ appContext: appContext }, results, callback);
```

Then in your plugin:

```
module.exports = function (context) {

   context.common(this); // do your common stuff here...

};
```

**Comment by alvarolorentedev** [comment]
yes but i am trying to do a crossplatform social network manager, so i need access to a plugin for example to post. 
That in my brain feels like a direct request from the app side towards the plugin itself. It feels not natural to do a double indirection to achieve it if i know all my plugins have the same signature. 
Thanks for the fast reply :+1:

**Comment by justinmchase** [comment]
Can you give a little code snippet with what you want to do with it?

**Comment by alvarolorentedev** [comment]
yes of course, first of course i will load the plugins and keep a reference to the collection

``` js
view.document.addEventListener('DOMContentLoaded', function () {
plugins.load(context, (err, ids, plugins) => {
                if(err)
                    return;
                this.pluginsHolder = plugins;
            });
});
```

each plugin contains a social network, each of the plugins export a class with the same signature. So i can do something like:

``` js
function postData()
{
       for (var i = 0; i < this.pluginsHolder.lenght; ++i) {
                this.pluginsHolder[i].post('Hello World');
       }
}
```

**Comment by justinmchase** [comment]
Oh ok! I'm glad you provided that snippet. You're just saying that you want the `modules` to also be added as a parameter to the callback?

e.g.

``` javascript
callback(null, dependencies, modules)
```

That would be pretty easy and backward compatible if you say yes I can publish this change pretty fast.

**Comment by alvarolorentedev** [comment]
yes that us exactly what i was thinking about :). if you publish that change it will be really helpful for me.

**Comment by justinmchase** [comment]
Published in [v0.1.0](https://github.com/EvolveLabs/electron-plugins/releases/tag/v0.1.0)

**Comment by alvarolorentedev** [comment]
thanks a lot :)
